src/basic/FStar.Errors.fs

```diff
@@ -868,12 +1021,12 @@ let add_errors errs =
 
 let issue_of_exn = function
     | Error(e, msg, r) ->
-      let errno = errno_of_error (e) in
+      let errno = error_number (lookup e) in
       Some (mk_issue EError (Some r) (message_prefix.append_prefix msg) (Some errno))
     | NYI msg ->
       Some (mk_issue ENotImplemented None (message_prefix.append_prefix msg) None)
```

Comment:
This seems like the only place where we put a `None` for the `issue_number`. But `NYI` is also never raised. Could we just kill it and make the `issue_number` an `int`? (anyway for after this PR). We could just default to some "Error_UnhandledExn" instead, with a given error number.

Reply:
Ok, I'll fix that in this PR removing NYI.

Reply:
Actually, FStar.Errors.diag sets the issue number to None for EInfo issues. So, I'm leaving the issue_number as an option, but am FStar.Util.NYI and FStar.Util.HardError ... neither are used.

Reply:
Oh sorry. Yeah nevermind, we can look at this later.